cli/gas.c

```diff
@@ -287,7 +287,7 @@ static int gas_dump(int argc, char **argv)
 	} cfg = {};
 	const struct argconfig_options opts[] = {
 		DEVICE_OPTION,
-		{"count", 'n', "NUM", CFG_LONG_SUFFIX, &cfg.count, required_argument,
+		{"count", 'n', "NUM", CFG_POSITIVE, &cfg.count, required_argument,
```

Comment:
This patch is completely wrong. CFG_LONG_SUFFIX accepts size arguments with a suffix. For example '4k' would parse as 4096. Positive, Non-negative, etc don't do this. I don't see a lot of value in changing this.

Reply:
Yes, I know, a suffix such as "Pi" will auto-complete the zeros after the size number.

But what we come across was some one input a negative value (example: for gas read "--addr/-a" argument), finally this negative is convert to a huge positive (CFG_LONG_SUFFIX type handler return value was "int", while "addr" variant is type of "unsigned long"), this is not what we expected.

I don't want to change CFG_LONG_SUFFIX handler to same as CFG_POSITIVE/CFG_NONNEGATIVE handler, namely treat it as error when negative value input.
On the hand, "trick" like "suffix" is unnecessary for these gas operations.
These are the reaons for me to change these argument type.

Reply:
No the 'suffix' is not a trick, it's a feature. I don't know what you mean by "Pi" or auto-complete zeros... that's really not what's happening. 

If there's a problem with LONG_SUFFIX, then fix it. But removing the suffix feature is not acceptable.